back-end/src/dms/models/books.py

```diff
@@ -57,3 +58,57 @@ class Meta:
 
     def __str__(self):
         return f"{self.user} - {self.book}"
+
+
+@receiver(models.signals.post_delete)
+def auto_delete_cover_on_book_delete(sender, instance: Book, **kwargs):
+    # pylint: disable=unused-argument
+    if not issubclass(sender, Book):
+        return
+
+    if instance.cover:
+        instance.cover.delete()
+
+
+@receiver(models.signals.pre_save)
+def auto_delete_cover_on_book_change(sender, instance: Book, **kwargs):
+    # pylint: disable=unused-argument
+    if not issubclass(sender, Book):
+        return
+    print("bk changed")
+
+    if not instance.pk:
+        return
+
+    try:
+        old_instance: Book = Book.objects.get(pk=instance.pk)
+    except Book.DoesNotExist:
+        return
+    if old_instance.cover != instance.cover and old_instance.cover:
+        old_instance.cover.delete()
+
+
+@receiver(models.signals.post_delete)
+def auto_delete_image_on_cover_delete(sender, instance: Cover, **kwargs):
+    # pylint: disable=unused-argument
+    if not issubclass(sender, Cover):
+        return
+
+    if instance and instance.image:
+        instance.image.delete(save=False)
+
+
+@receiver(models.signals.pre_save)
+def auto_delete_image_on_cover_change(sender, instance: Cover, **kwargs):
+    # pylint: disable=unused-argument
+
+    if not issubclass(sender, Cover):
+        return
+    if not instance.pk:
+        return
+    try:
+        old_instance: Cover = Cover.objects.get(pk=instance.pk)
+    except Cover.DoesNotExist:
+        return
+    if old_instance.image != instance.image and instance.image:
+        old_instance.image.delete(save=False)
```

Comment:
When does the `Cover` change? 
Maybe we should delete the `Cover` instance too since it is a `OneToOneField`?

Reply:
"`Cover` change" here means that `class Cover`'s field `image` is changed to another image. It changes every time a new book cover is attached, so this code should be useful.